sycl/test/warnings/sycl_121_deprecations.cpp

```diff
@@ -0,0 +1,63 @@
+// RUN: %clangxx -fsycl -sycl-std=2020 -Xclang -verify -Xclang -verify-ignore-unexpected=note %s -o %t.out
+// RUN: %clangxx -fsycl -Xclang -verify -Xclang -verify-ignore-unexpected=note %s -o %t.out
+// RUN: %clangxx -fsycl -sycl-std=2017 -Werror %s -o %t.out
+// RUN: %clangxx -fsycl -sycl-std=1.2.1 -Werror %s -o %t.out
+
+#include <CL/sycl.hpp>
+
+int main() {
+  cl_context ClCtx;
+  // expected-warning@+1 {{'context' is deprecated: OpenCL interop APIs are deprecated}}
+  sycl::context Ctx{ClCtx};
+  // expected-warning@+1 {{'get' is deprecated: OpenCL interop APIs are deprecated}}
+  (void)Ctx.get();
+
+  cl_mem Mem;
+  // expected-warning@+1 {{'buffer' is deprecated: OpenCL interop APIs are deprecated}}
+  sycl::buffer<int, 1> Buf{Mem, Ctx};
+  (void)Buf.get_size();
```

Comment:
What's the point of calling `get_size()` here (or `get_addressing_mode` for sampler)? Is `(void)Buf` not enough here?